code/game/machinery/porta_turret/portable_turret.dm

```diff
@@ -413,17 +400,17 @@
 			else if(check_anomalies) //non humans who are not simple animals (xenos etc)
 				if(!in_faction(C))
 					targets += C
-
-		if(ismecha(A))
-			var/obj/mecha/M = A
-			//If there is a user and they're not in our faction
-			if(M.occupant && !in_faction(M.occupant))
-				if(assess_perp(M.occupant) >= 4)
-					targets += M
-
-	if(!tryToShootAt(targets))
-		if(!always_up)
-			popDown() // no valid targets, close the cover
+	for(var/A in GLOB.mechas_list)

```

Comment:
why is having to loop through the global list and check if the turret can see them faster than checking for mecha types inrange?

Reply:
yes, this is now two loops instead of one, although admittedly there should be very many mechs at any time.

Reply:
the answer is, `view()` is extremely slow, untyped view like it was previously is even slower and fills the loop with garbage, like "oh boy i sure love checking if 800 lighting objects are mechas" tier.

Reply:
Smart